### ISSUE TITLE
Support for .org files markup

### ISSUE THREAD

**Issue by jackjackk** [gogits/gogs#2809]
as it is done in github (https://github.com/github/markup).
would it be possible/feasible?

thanks
Giacomo

**Comment by j4** [comment]
I'm voting for !

**Comment by holocronweaver** [comment]
What would be involved in implementing this? Based on amount of work, I may be willing to implement it.

I see there is a [org-mode parser for Go](https://github.com/jacobmoe/gorg).

**Comment by Unknwon** [comment]
Patch has pushed to fix this issue, please test on `develop` branch or https://try.gogs.io.

**Comment by holocronweaver** [comment]
Results in a blank page for me: https://try.gogs.io/holocronweaver/org-test

The only thing in that repo is a `README.org` file.

**Comment by Unknwon** [comment]
@holocronweaver thanks for the report! Looks like many edge cases haven't been covered by the parser. I only tested https://raw.githubusercontent.com/Digit/witch/master/birch/README.org. 

Would you mind comment the raw content that cause problem to the thread https://github.com/chaseadamsio/goorgeous/issues/37?

**Comment by Unknwon** [comment]
Nvm, extracted from demo site.

**Comment by rogersachan** [comment]
What about files that don't use the org ending but still invoke as said here https://orgmode.org/manual/Activation.html ?

**Comment by Unknwon** [comment]
@rogersachan could do same thing as Markdown here https://github.com/gogs/gogs/blob/master/conf/app.ini#L137-L145

**Comment by Ambrevar** [comment]
I find that .org files work on notabug:

https://notabug.org/Ambrevar/mu4e-conversation/src/master/readme.org

Most of it works, although there are some issues still.  For instance I can't seem to be able to display images inline.
(See https://notabug.org/Ambrevar/mu4e-conversation/src/master/readme.org for the syntax.)

What about leaving the job to Emacs: use Org mode to export to HTML and display that HTML?

**Comment by thierryvolpiatto** [comment]
Would be great to handle org files in wikis as well, see issue https://notabug.org/hp/gogs/issues/175
Thanks.

**Comment by Unknwon** [comment]
@Ambrevar 

>What about leaving the job to Emacs: use Org mode to export to HTML and display that HTML?

Even though I'm not sure what you mean, but seems like require user to install Emacs on the server machine.

@thierryvolpiatto: I think it is reasonable, I've adjust the issue title.

**Comment by Ambrevar** [comment]
Yes, that's what I meant: install Emacs on the server machine and let it export from .org to .html.  Emacs can be used in "batch" mode to run scripts.

**Comment by Ambrevar** [comment]
Because pandoc's Org support is only partial.
It will often fail on simple syntax such as `~` and `=` (for code snippets, which is way too common).
